### PR TITLE
add CSV Source summary stats to audit log

### LINKED ISSUE #675: Add ability to print or export a list of all CVR files added to a config
Also requested that this report show the number of ballots appearing in each CVR. 

I see three ways to do this: 

1) Include this data in the audit.log
2) Create a separate file upon tabulation listing each CVR file and the number of ballots read in from those files
3) Allow creation of this file before tabulation (but would require parsing all CVRs - is that worth worrying about?)

### PR DESCRIPTION
**Pull request by artoonie**
closes #675

### CODE REVIEW

File: src/main/java/network/brightspots/rcv/ResultsWriter.java [hunk 1]
```diff
@@ -1058,15 +1058,22 @@ static class PerSourceDataForCsv {
...
+        CvrSource source, BaseCvrReader reader,
+        int sourceIndex, int firstIndexInCvrList, int lastIndexInCvrList) {
       this.source = source;
       this.reader = reader;
       this.sourceIndex = sourceIndex;
+      this.firstIndexInCvrList = firstIndexInCvrList;
       this.lastIndexInCvrList = lastIndexInCvrList;
     }
+
+    public int getNumCvrs() {
```
[the remark this answers is not in the document]
Comment (artoonie) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java:1074]: I was curious why the styleguide didn't check this -- turns out there's a minimum line count to require javadoc on a public method (n=2). Should I remove that filter?

File: src/main/java/network/brightspots/rcv/TabulatorSession.java [hunk 2]
```diff
@@ -356,6 +358,9 @@ private List<CastVoteRecord> parseCastVoteRecords(ContestConfig config) {
       // error already logged in ResultsWriter
     }
 
+    // Print contest summary to the audit log
+    writeContestSummary(perSourceDataForCsv);
```
[the remark this answers is not in the document]
Comment (artoonie) [reply src/main/java/network/brightspots/rcv/TabulatorSession.java:391]: I'm realizing now that this PR has a ton of overlap with #835. One is designed for the GUI, one for the audit log. Let me explore combining them into a single PR, and that should accomplish this comment too.

File: src/main/java/network/brightspots/rcv/ResultsWriter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java:1078]: Meh, probably fine to leave it. We can always go back later and fix if someone complains. Meaning it's probably also ok to leave this uncommented I guess.

**Comment by artoonie** [comment]
I've merged this into #835 and I'm going to close out this PR. Totally forgot that they do nearly duplicate work, but one prints to the audit log and the other to the interface.

**Pull request closed without merging**